woocommerce/woocommerce-android
# Product Creation with AI: saving product + navigation

Closes: #9832 

### Description
This PR implements two parts:
1. Handles saving the generated product as draft.
2. Navigation to the product details screen with the received order id.

### Testing instructions
1. Use an AI eligible store.
2. Open the app and start the AI product creation.
3. Complete the flow.
4. Confirm the Save as draft button is shown.
5. Click on it.
6. Confirm a loading indicator is shown, then after finishing that app navigates to the product details screen.
7. Confirm the product details are correct.
8. Navigate back, and confirm the correct behavior.

### Images/gif
https://github.com/woocommerce/woocommerce-android/assets/1657201/495a87f0-2513-42a5-9ae1-d8fc6946151f
